src/core/src/op/assign.cpp

```diff
@@ -33,93 +31,95 @@ void op::v3::Assign::validate_and_infer_types() {
 
     set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
 }
 
-shared_ptr<Node> op::v6::Assign::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> Assign::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v6_Assign_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return std::make_shared<op::v6::Assign>(new_args.at(0), m_variable);
+    return std::make_shared<Assign>(new_args.at(0), m_variable);
 }
 
-bool op::v6::Assign::visit_attributes(AttributeVisitor& visitor) {
+bool Assign::visit_attributes(AttributeVisitor& visitor) {
     OV_OP_SCOPE(v6_Assign_visit_attributes);
     visitor.on_attribute("variable_id", m_variable);
     return true;
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-bool op::v6::Assign::evaluate(const HostTensorVector& outputs,
-                              const HostTensorVector& inputs,
-                              const EvaluationContext& evaluation_context) const {
+bool Assign::evaluate(TensorVector& outputs,
+                      const TensorVector& inputs,
+                      const EvaluationContext& evaluation_context) const {
     OV_OP_SCOPE(v6_Assign_evaluate);
     const auto& found_context = evaluation_context.find("VariableContext");
     NODE_VALIDATION_CHECK(this, found_context != evaluation_context.end(), "VariableContext not found.");
 
-    auto& variable_context = const_cast<VariableContext&>(found_context->second.as<VariableContext>());
+    auto& variable_context = const_cast<util::VariableContext&>(found_context->second.as<util::VariableContext>());
 
     const auto& variable_values = variable_context.get_variable_values();
 
     // automatically allocate memory if not provided by user
     if (variable_values.find(m_variable) == variable_values.end()) {
-        auto host_tensor =
-            std::make_shared<ngraph::HostTensor>(m_variable->get_info().data_type, m_variable->get_info().data_shape);
-        variable_context.set_variable_value(m_variable, make_shared<VariableValue>(host_tensor));
+        auto tensor = Tensor(m_variable->get_info().data_type, m_variable->get_info().data_shape.to_shape());
+        variable_context.set_variable_value(m_variable, std::make_shared<util::VariableValue>(tensor));
     }
 
     const auto var_value = variable_values.find(m_variable)->second;
     var_value->set_reset(false);
-    const auto& buffer = var_value->get_value();
-    buffer->set_unary(inputs[0]);
-    outputs[0]->set_unary(inputs[0]);
+    auto buffer = var_value->get_state();
+    buffer.set_shape(inputs[0].get_shape());
+    outputs[0].set_shape(inputs[0].get_shape());
 
-    void* input = inputs[0]->get_data_ptr();
-    outputs[0]->write(input, outputs[0]->get_size_in_bytes());
-    buffer->write(input, buffer->get_size_in_bytes());
+    inputs[0].copy_to(outputs[0]);
+    inputs[0].copy_to(buffer);
```

Comment:
@ilyachur 
```suggestion
    std::memcpy(outputs[0].data(), inputs[0].data(), inputs[0].get_byte_size());
    std::memcpy(buffer.data(), inputs[0].data(), inputs[0].get_byte_size());
```
Or use mem copy is sufficient here (it produce smaller binary size)?

Reply:
I think memcpy will be enough for this case, `copy_to` is a sugar in order to make a copy of tensor to new tensor with different layout.
As I understand, here inputs and outputs has the same layouts and shapes

Reply:
OK, thanks for explanation. Will change to `memcpy`